src/AutoRest.CSharp/LowLevel/Output/DpgOutputLibraryBuilder.cs

```diff
@@ -276,6 +276,35 @@ private IReadOnlyList<ClientInfo> SetHierarchy(IReadOnlyDictionary<string, Clien
             return new[] { topLevelClientInfo };
         }
 
+        private static void AssignParents(in IEnumerable<InputClient> clients, IReadOnlyDictionary<string, ClientInfo> clientInfosByName)
+        {
+            foreach (var client in clients)
+            {
+                if (!String.IsNullOrEmpty(client.Parent))
+                {
+                    ClientInfo? targetClient = null;
+                    ClientInfo? targetParent = null;
+                    foreach (var info in clientInfosByName.Values)
+                    {
+                        if (info.OperationGroupKey == client.Name)
+                            targetClient = info;
+                        if (info.OperationGroupKey == client.Parent)
+                            targetParent = info;
+                    }
+                    if (targetClient != null && targetParent != null)
+                    {
+                        targetClient.Parent = targetParent;
+                        targetParent.Children.Add(targetClient);
+                        if (targetParent.ClientParameters.Count == 0)
+                        {
+                            var endpointParameter = targetParent.Children.SelectMany(c => c.ClientParameters).FirstOrDefault(p => p.IsEndpoint);
+                            var clientParameters = endpointParameter != null ? new[] { endpointParameter } : Array.Empty<InputParameter>();
+                            targetParent.ClientParameters = clientParameters;
```

Comment:
If this is the only place that we should be setting this `ClientParameters` property, we should do it in its getter instead of here

Reply:
Thanks @ArcturusZhang  updated